genienlp/train.py

```diff
@@ -228,12 +222,35 @@ def should_log(iteration, log_every):
     return iteration % log_every == 0
 
 
-def do_validate(
-    iteration, args, model, numericalizer, val_iters, *, train_task, round_progress, task_progress, writer, logger
-):
+def validate(task, val_iter, model, args, num_print=10):
```

Comment:
The new changes to the validation code have definitely made it cleaner. Can you change the naming of `validate()` and `do_validate()` in train.py to `validate_during_training()` and `do_validate_during_training()` to further differentiate them from `validate()` defined elsewhere?

Reply:
Validate and evaluate are often used interchangeably, so doesn't help with confusion. I just think validate_during_training() is more descriptive, and e.g. made me quickly realize I should not use it for the new prediction code I plan to write.

Reply:
Ok let's go with your suggestion